Plutarch/Bool.hs

```diff
@@ -117,3 +139,37 @@ por = phoistAcyclic $ plam $ \x y -> pif' # x # (phoistAcyclic $ pdelay $ pcon P
 -- | Hoisted, Plutarch level, strictly evaluated boolean or function.
 por' :: Term s (PBool :--> PBool :--> PBool)
 por' = phoistAcyclic $ plam $ \x y -> pif' # x # (pcon PTrue) # y
+
+-- | Generic version of (#==)
+gpeq ::
+  forall t s.
+  ( PGeneric s t
+  , PlutusType t
+  , All2 PEq (PCode s t)
+  ) =>
+  Term s (t :--> t :--> PBool)
+gpeq =
+  phoistAcyclic $
+    punsafeAsClosedTerm @s $
+      plam $ \x y ->
+        pmatch x $ \x' ->
```

Comment:
since chained `pmatch`es like these are really inefficient (and since we have no other way to do this right now) - is it worth thinking about implementing a better method, in the future?